### DISCUSSION TITLE
legacyupdate.net | 526: Invalid SSL certificate

### DISCUSSION THREAD

**Discussion by Chuterix** [LegacyUpdate/LegacyUpdate#66] [Questions & Help] [answered]
Invalid SSL certificate Error code 526
Visit [cloudflare.com](https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net) for more information.
2022-12-23 21:44:36 UTC
You
Browser
Working
Chicago
Cloudflare
Working
legacyupdate.net
Host
Error
What happened?

The origin web server does not have a valid SSL certificate.
What can I do?
If you're a visitor of this website:

Please try again in a few minutes.
If you're the owner of this website:

The SSL certificate presented by the server did not pass validation. This could indicate an expired SSL certificate or a certificate that does not include the requested domain name. Please contact your hosting provider to ensure that an up-to-date and valid SSL certificate issued by a Certificate Authority is configured for this domain name on the origin server. [Additional troubleshooting information here.](https://support.cloudflare.com/hc/en-us/articles/200721975)

Cloudflare Ray ID: 77e447891a68e1ff • Your IP:

• Performance & security by [Cloudflare](https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net)

`<!DOCTYPE html>

 <html class="no-js" lang="en-US"> 
<head>

<title>legacyupdate.net | 526: Invalid SSL certificate</title>
<meta charset="UTF-8" />
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<meta http-equiv="X-UA-Compatible" content="IE=Edge" />
<meta name="robots" content="noindex, nofollow" />
<meta name="viewport" content="width=device-width,initial-scale=1" />
<link rel="stylesheet" id="cf_styles-css" href="[/cdn-cgi/styles/main.css](view-source:https://legacyupdate.net/cdn-cgi/styles/main.css)" />

</head>
<body>
<div id="cf-wrapper">

    

    <div id="cf-error-details" class="p-0">
        <header class="mx-auto pt-10 lg:pt-6 lg:px-8 w-240 lg:w-full mb-8">
            <h1 class="inline-block sm:block sm:mb-2 font-light text-60 lg:text-4xl text-black-dark leading-tight mr-2">
              
              <span class="inline-block">Invalid SSL certificate</span>
              <span class="code-label">Error code 526</span>
            </h1>
            <div>
               Visit <a href="[https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net](view-source:https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net)" target="_blank" rel="noopener noreferrer">cloudflare.com</a> for more information.
            </div>
            <div class="mt-3">2022-12-23 21:44:36 UTC</div>
        </header>
        
        <div class="my-8 bg-gradient-gray">
            <div class="w-240 lg:w-full mx-auto">
                <div class="clearfix md:px-8">
                  
<div id="cf-browser-status" class=" relative w-1/3 md:w-full py-15 md:p-0 md:py-8 md:text-left md:border-solid md:border-0 md:border-b md:border-gray-400 overflow-hidden float-left md:float-none text-center">
  <div class="relative mb-10 md:m-0">
    
    <span class="cf-icon-browser block md:hidden h-20 bg-center bg-no-repeat"></span>
    <span class="cf-icon-ok w-12 h-12 absolute left-1/2 md:left-auto md:right-0 md:top-0 -ml-6 -bottom-4"></span>
    
  </div>
  <span class="md:block w-full truncate">You</span>
  <h3 class="md:inline-block mt-3 md:mt-0 text-2xl text-gray-600 font-light leading-1.3">
    
    Browser
    
  </h3>
  <span class="leading-1.3 text-2xl text-green-success">Working</span>
</div>

<div id="cf-cloudflare-status" class=" relative w-1/3 md:w-full py-15 md:p-0 md:py-8 md:text-left md:border-solid md:border-0 md:border-b md:border-gray-400 overflow-hidden float-left md:float-none text-center">
  <div class="relative mb-10 md:m-0">
    <a href="[https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net](view-source:https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net)" target="_blank" rel="noopener noreferrer">
    <span class="cf-icon-cloud block md:hidden h-20 bg-center bg-no-repeat"></span>
    <span class="cf-icon-ok w-12 h-12 absolute left-1/2 md:left-auto md:right-0 md:top-0 -ml-6 -bottom-4"></span>
    </a>
  </div>
  <span class="md:block w-full truncate">Chicago</span>
  <h3 class="md:inline-block mt-3 md:mt-0 text-2xl text-gray-600 font-light leading-1.3">
    <a href="[https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net](view-source:https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net)" target="_blank" rel="noopener noreferrer">
    Cloudflare
    </a>
  </h3>
  <span class="leading-1.3 text-2xl text-green-success">Working</span>
</div>

<div id="cf-host-status" class="cf-error-source relative w-1/3 md:w-full py-15 md:p-0 md:py-8 md:text-left md:border-solid md:border-0 md:border-b md:border-gray-400 overflow-hidden float-left md:float-none text-center">
  <div class="relative mb-10 md:m-0">
    
    <span class="cf-icon-server block md:hidden h-20 bg-center bg-no-repeat"></span>
    <span class="cf-icon-error w-12 h-12 absolute left-1/2 md:left-auto md:right-0 md:top-0 -ml-6 -bottom-4"></span>
    
  </div>
  <span class="md:block w-full truncate">legacyupdate.net</span>
  <h3 class="md:inline-block mt-3 md:mt-0 text-2xl text-gray-600 font-light leading-1.3">
    
    Host
    
  </h3>
  <span class="leading-1.3 text-2xl text-red-error">Error</span>
</div>

                </div>
              
            </div>
        </div>

        <div class="w-240 lg:w-full mx-auto mb-8 lg:px-8">
            <div class="clearfix">
                <div class="w-1/2 md:w-full float-left pr-6 md:pb-10 md:pr-0 leading-relaxed">
                    <h2 class="text-3xl font-normal leading-1.3 mb-4">What happened?</h2>
                    <p>The origin web server does not have a valid SSL certificate.</p>
                </div>
              
                <div class="w-1/2 md:w-full float-left leading-relaxed">
                    <h2 class="text-3xl font-normal leading-1.3 mb-4">What can I do?</h2>
                          <h3 class="text-15 font-semibold mb-2">If you're a visitor of this website:</h3>
      <p class="mb-6">Please try again in a few minutes.</p>

      <h3 class="text-15 font-semibold mb-2">If you're the owner of this website:</h3>
      <p><span>The SSL certificate presented by the server did not pass validation. This could indicate an expired SSL certificate or a certificate that does not include the requested domain name. Please contact your hosting provider to ensure that an up-to-date and valid SSL certificate issued by a Certificate Authority is configured for this domain name on the origin server.</span> <a rel="noopener noreferrer" href="[https://support.cloudflare.com/hc/en-us/articles/200721975](view-source:https://support.cloudflare.com/hc/en-us/articles/200721975)">Additional troubleshooting information here.</a></p>
                </div>
            </div>
              
        </div>

        <div class="cf-error-footer cf-wrapper w-240 lg:w-full py-10 sm:py-4 sm:px-8 mx-auto text-center sm:text-left border-solid border-0 border-t border-gray-300">
  <p class="text-13">
    <span class="cf-footer-item sm:block sm:mb-1">Cloudflare Ray ID: <strong class="font-semibold">77e447891a68e1ff</strong></span>
    <span class="cf-footer-separator sm:hidden">&bull;</span>
    <span id="cf-footer-item-ip" class="cf-footer-item hidden sm:block sm:mb-1">
      Your IP:
      <button type="button" id="cf-footer-ip-reveal" class="cf-footer-ip-reveal-btn">Click to reveal</button>
      <span class="hidden" id="cf-footer-ip">removed</span>
      <span class="cf-footer-separator sm:hidden">&bull;</span>
    </span>
    <span class="cf-footer-item sm:block sm:mb-1"><span>Performance &amp; security by</span> <a rel="noopener noreferrer" href="[https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net](view-source:https://www.cloudflare.com/5xx-error-landing?utm_source=errorcode_526&utm_campaign=legacyupdate.net)" id="brand_link" target="_blank">Cloudflare</a></span>
    
  </p>
  <script>(function(){function d(){var b=a.getElementById("cf-footer-item-ip"),c=a.getElementById("cf-footer-ip-reveal");b&&"classList"in b&&(b.classList.remove("hidden"),c.addEventListener("click",function(){c.classList.add("hidden");a.getElementById("cf-footer-ip").classList.remove("hidden")}))}var a=document;document.addEventListener&&a.addEventListener("DOMContentLoaded",d)})();</script>
</div>

    </div>
</div>
</body>
</html>`

**Comment by ismaelbalaghni** [comment]
I'd suggest you to clean a little bit your message, it contains your IP address. However, I am facing the same issue and I think this is what prevents me to make Legacy Update work on Vista.

We will have to wait as it may be a Cloudflare bug? When using the website recommended by Cloudflare to check the validity of the SSL certificate and I almost get the same results every time: all is correct.

**Reply by kirb** [reply]
The issue was that the certificate on my server, which is used for the communication between Cloudflare and Legacy Update, was failing to automatically renew. Since I have strict certificate checking enabled with Cloudflare, their servers were rejecting to pass the traffic along to my server, even though the communication between yourself and Cloudflare was fine.

Indeed, Legacy Update uses https on Vista, so it would have affected that. XP would have been fine since it has to fall back to plain http there (#3).

**Comment by kirb** [comment, marked as answer]
Thanks, fixed now. More info here: https://github.com/kirb/LegacyUpdate/discussions/68
